contracts/interchain-token-service/src/contract.rs

```diff
@@ -146,17 +169,52 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     }
 
     fn deploy_interchain_token(
-        _env: &Env,
-        _caller: Address,
-        _token_id: String,
-        _source_address: Bytes,
-        _destination_chain: String,
-        _destination_address: Bytes,
-        _amount: i128,
-        _metadata: Bytes,
+        env: &Env,
+        caller: Address,
+        salt: BytesN<32>,
+        name: String,
+        symbol: String,
+        decimal: u32,
+        initial_supply: i128,
+        minter: Address,
         _gas_token: Token,
     ) {
-        todo!()
+        caller.require_auth();
+
+        // TODO: minterBytes
+        let _minter_bytes = minter;
+
+        let _deploy_salt = Self::interchain_token_deploy_salt(env, caller, salt);
+
+        let _token_meta_data = TokenMetadata {
+            decimal,
+            name,
+            symbol,
+        };
+
+        let _token_id = BytesN::<32>::from_array(env, &[0; 32]);
+
+        // TODO: _deployInterchainToken(deploySalt, currentChain, name, symbol, decimals, minterBytes, gasValue);
+
+        if initial_supply > 0 {
+            // TODO: mint, transferMintership
+        }
+    }
+
+    fn interchain_token_deploy_salt(env: &Env, deployer: Address, salt: BytesN<32>) -> BytesN<32> {
```

Comment:
avoid changing the other interfaces in this PR to reduce conflicts with Attiss's PR. you can add this method and unit test it for now